### PR TITLE
update dependencies for myproxyclient

### LINKED ISSUE #12: Missing dependencies myproxyclient and defusedxml
- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

When I install the `esgf-pyclient` package from conda-forge and try to use the logon module, it crashes:
```python
>>> import pyesgf.logon
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/bandela/conda/envs/pyesgf-test/lib/python3.9/site-packages/pyesgf/logon.py", line 42, in <module>
    from defusedxml import ElementTree
ModuleNotFoundError: No module named 'defusedxml'
```

It looks like the `defusedxml` and `myproxyclient` packages, listed in [requirements.txt](https://github.com/ESGF/esgf-pyclient/blob/master/requirements.txt) are missing from https://github.com/conda-forge/esgf-pyclient-feedstock/blob/1047da2f484174903a08a29bb81709804d4dffa5/recipe/meta.yaml#L21-L27
and the use of the pyesgf.logon module is not tested
https://github.com/conda-forge/esgf-pyclient-feedstock/blob/1047da2f484174903a08a29bb81709804d4dffa5/recipe/meta.yaml#L28-L33

Is this on purpose? In the documentation it is [mentioned](https://esgf-pyclient.readthedocs.io/en/latest/notebooks/examples/logon.html) that I would need to install `myproxyclient` from PyPI if I want to use the `pyesgf.logon` module, but there is no mention of `defusedxml`. Also, an installation from PyPI automatically includes both `myproxyclient` and `defusedxml`.

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /home/bandela/conda/envs/pyesgf-test:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
brotlipy                  0.7.0           py39h3811e60_1001    conda-forge
ca-certificates           2021.5.30            ha878542_0    conda-forge
certifi                   2021.5.30        py39hf3d152e_0    conda-forge
cffi                      1.14.6           py39he32792d_0    conda-forge
chardet                   4.0.0            py39hf3d152e_1    conda-forge
charset-normalizer        2.0.0              pyhd8ed1ab_0    conda-forge
cryptography              3.4.7            py39hbca0aa6_0    conda-forge
esgf-pyclient             0.3.0              pyh44b312d_0    conda-forge
idna                      3.1                pyhd3deb0d_0    conda-forge
jinja2                    3.0.1              pyhd8ed1ab_0    conda-forge
ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
libffi                    3.3                  h58526e2_2    conda-forge
libgcc-ng                 11.1.0               hc902ee8_8    conda-forge
libgomp                   11.1.0               hc902ee8_8    conda-forge
libstdcxx-ng              11.1.0               h56837e0_8    conda-forge
markupsafe                2.0.1            py39h3811e60_0    conda-forge
ncurses                   6.2                  h58526e2_4    conda-forge
openssl                   1.1.1k               h7f98852_1    conda-forge
pip                       21.2.4             pyhd8ed1ab_0    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pyopenssl                 20.0.1             pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1            py39hf3d152e_3    conda-forge
python                    3.9.6           h49503c6_1_cpython    conda-forge
python_abi                3.9                      2_cp39    conda-forge
readline                  8.1                  h46c0cb4_0    conda-forge
requests                  2.26.0             pyhd8ed1ab_0    conda-forge
requests_cache            0.4.13                     py_0    conda-forge
setuptools                57.4.0           py39hf3d152e_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
sqlite                    3.36.0               h9cd32fc_0    conda-forge
tk                        8.6.10               h21135ba_1    conda-forge
tzdata                    2021a                he74cb21_1    conda-forge
urllib3                   1.26.6             pyhd8ed1ab_0    conda-forge
webob                     1.8.7              pyhd8ed1ab_0    conda-forge
wheel                     0.37.0             pyhd8ed1ab_1    conda-forge
xz                        5.2.5                h516909a_1    conda-forge
zlib                      1.2.11            h516909a_1010    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

     active environment : pyesgf-test
    active env location : /home/bandela/conda/envs/pyesgf-test
            shell level : 2
       user config file : /home/bandela/.condarc
 populated config files : /home/bandela/.condarc
          conda version : 4.10.3
    conda-build version : 3.20.5
         python version : 3.8.3.final.0
       virtual packages : __linux=5.11.0=0
                          __glibc=2.33=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/bandela/conda  (writable)
      conda av data dir : /home/bandela/conda/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/bandela/conda/pkgs
                          /home/bandela/.conda/pkgs
       envs directories : /home/bandela/conda/envs
                          /home/bandela/.conda/envs
               platform : linux-64
             user-agent : conda/4.10.3 requests/2.25.1 CPython/3.8.3 Linux/5.11.0-25-generic ubuntu/21.04 glibc/2.33
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False

```
</details>

### PR DESCRIPTION
**Pull request by cehbrecht**
Fix #12

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by cehbrecht** [comment]
@conda-forge-admin, please rerender

**Comment by cehbrecht** [comment]
@bouweandela do you like to be added as maintainer for the conda package?

**Comment by bouweandela** [comment]
Sure, I'd be happy to help out!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,6 +24,8 @@ requirements:
     - requests
     - requests_cache
     - webob
+    - defusedxml
+    - myproxyclient
 
 test:
   imports:
```
Comment (bouweandela) [inline recipe/meta.yaml:31]: Maybe you could add an import of `pyesgf.logon` as a test here?

Comment (cehbrecht) [reply recipe/meta.yaml:31]: done

**Pull request merged**